cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1762,6 +1762,41 @@ const char* replace_utf8_utf8_utf8(gdv_int64 context, const char* text,
                                              out_len);
 }
 
+// Returns the quoted string (Includes escape character for any single quotes)
+// E.g. DONT  -> 'DONT'
+//      DON'T -> 'DON\'T'
+FORCE_INLINE
+const char* quote_utf8(gdv_int64 context, const char* in, gdv_int32 in_len,
+                       gdv_int32* out_len) {
+  if (in_len <= 0) {
+    *out_len = 0;
+    return "";
+  }
+  // try to allocate double size output string (worst case)
+  auto out = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, in_len * 2));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+  // The output string should start with a single quote
+  out[0] = '\'';
+  gdv_int32 counter = 1;
+  for (int i = 0; i < in_len; i++) {
+    if (memcmp(in + i, "'", 1) == 0) {
+      out[counter] = '\\';
+      counter++;
+      out[counter] = '\'';
+    } else {
+      out[counter] = in[i];
+    }
+    counter++;
+  }
+  out[counter] = '\'';
```

Comment:
won't this overflow for the case we have all quotes. Say we have 2 quotes and in_len is 2.. we allocate only 4 bytes, but this logic may need 5 bytes as follows '\'\'' which is 5 chars and not 4.